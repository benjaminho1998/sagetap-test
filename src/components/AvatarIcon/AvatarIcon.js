import React from 'react';
import Avatar from '@mui/material/Avatar';

const AvatarIcon = (props) => {

    //This function is from https://mui.com/components/avatars/
    //It chooses a different color for the avatar depending on the name
    const stringToColor = (string) => {
        let hash = 0;
        let i;

        for (i = 0; i < string.length; i += 1) {
            hash = string.charCodeAt(i) + ((hash << 5) - hash);
        }

        let color = '#';

        for (i = 0; i < 3; i += 1) {
            const value = (hash >> (i * 8)) & 0xff;
            color += `00${value.toString(16)}`.substr(-2);
        }
        return color;
    }

    //This function is from https://mui.com/components/avatars/
    //It gives each avatar the css styles to change the background color
    const stringAvatar = (name) => {
        return {
            sx: {
                bgcolor: stringToColor(name),
            },
            children: `${name.split(' ')[0][0]}${name.split(' ')[1][0]}`,
        };
    }

    return (
        <Avatar style={{width: '45px', height: '45px'}} {...stringAvatar(props.name)} />
    );
}

export default AvatarIcon;