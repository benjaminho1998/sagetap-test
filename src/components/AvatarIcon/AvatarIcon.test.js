import React from 'react';
import renderer from 'react-test-renderer';
import AvatarIcon from './AvatarIcon';

describe('AvatarIcon Component', () => {
  it('renders snapshot correctly with props', () => {
    const name = 'Benjamin Ho';
    const tree = renderer.create(<AvatarIcon name={name} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
