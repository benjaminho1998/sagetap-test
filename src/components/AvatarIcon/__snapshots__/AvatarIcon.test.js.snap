// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarIcon Component renders snapshot correctly with props 1`] = `
<div
  className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-ln2kiz-MuiAvatar-root"
  style={
    Object {
      "height": "45px",
      "width": "45px",
    }
  }
>
  BH
</div>
`;
